templates/server-statefulset.yaml

```diff
@@ -63,6 +63,10 @@ spec:
     {{- if .Values.server.tolerations }}
       tolerations:
         {{ tpl .Values.server.tolerations . | nindent 8 | trim }}
+    {{- end }}
+    {{- if .Values.server.topologySpreadConstraints }}
```

Comment:
Could you add a check here similar to:
`{{- if or ( gt .Capabilities.KubeVersion.Major "1" ) ( ge .Capabilities.KubeVersion.Minor "18" ) }}` ?

There is a warning in the values file, but this might be helpful in ensuring it doesn't accidentally get templated when say re-using a values file. We have tests that do something similar in `ui-ingress.bats` for reference!

Reply:
I have added the checks.

A side note: do you think it's better to just have this fail instead of being silently ignored when a user tries to use this on a cluster < 1.18 though?

Reply:
YES! I like this idea a lot. Would you mind updating the PR to fail if the user attempts to set this value?

Reply:
I have added that in.